### DISCUSSION TITLE
More playback controls in replays

### DISCUSSION THREAD

**Discussion by LeNitrous** [ppy/osu#26444] [Ideas] [answered]
This stemmed from the fact that pausing and unpausing is only possible in desktop or mobile devices with a keyboard plugged in. Not all mobile devices have a keyboard plugged in and for convenience's sake, the empty playback control settings could have more buttons for more controlled seeking (as seeking in mobile not to mention with a small screen is difficult) and pausing/playing the replay.

![image](https://github.com/ppy/osu/assets/20495991/8efb8342-4a93-4cd6-a240-72845fe9031c)
In order: -10s, -5s, Play/Pause, +5s, +10s

As an aside, it would've been great to be able to seek frame-by-frame for more in-depth visual replay analysis but this is as far as I can go as I can't get around the `FrameStabilityContainer` inside a `DrawableRuleset`.

**Comment by peppy** [comment]
Sounds good to have. I'd also want to see the valid playback speed to allow lower speeds than currently possible.

**Comment by LeNitrous** [comment]
I've done a little POC on what "replay frame-based" seeking would look like. It's a big help analyzing player behavior on complex and fast sections of a map where even slowing down wouldn't probably help.

https://github.com/ppy/osu/assets/20495991/cb62b436-e37a-408a-bee0-be25552ab530

It changes something big with `MasterGameplayClockContainer` and `FrameStabilityContainer` so it's not something easy to add and will take a lot of considerations I assume.

**Comment by frenzibyte** [comment, marked as answer]
Long since been added, see https://github.com/ppy/osu/pull/26446.
